src/pytest_elasticsearch/factories.py

```diff
@@ -108,26 +157,32 @@ def elasticsearch_proc_fixture(request):
         home_path = os.path.join(
             tmpdir, 'elasticsearch_{0}'.format(elasticsearch_port))
         work_path = '{0}_tmp'.format(home_path)
+        conf_path = configuration_path or config['configuration_path']
 
         if discovery_zen_ping_multicast_enabled is not None:
             multicast_enabled = str(
                 discovery_zen_ping_multicast_enabled).lower()
         else:
             multicast_enabled = config['discovery_zen_ping_multicast_enabled']
 
-        command_exec = '''
-            {deamon} -p {pidfile} --http.port={port}
-            --path.home={home_path}  --default.path.logs={logs_path}
-            --default.path.work={work_path}
-            --default.path.conf=/etc/elasticsearch
-            --cluster.name={cluster}
-            --network.publish_host='{network_publish_host}'
-            --discovery.zen.ping.multicast.enabled={multicast_enabled}
-            --index.store.type={index_store_type}
-            '''.format(
+        exact = '{major}.{minor}.{patch}'.format(**version_parts)
+        major_minor = '{major}.{minor}'.format(**version_parts)
+        major = '{major}'.format(**version_parts)
+
+        # try exact version if there is such; otherwise fallback to minor;
+        # as final step fallback to major
+        command = commands_exec.get(exact) or \
```

Comment:
condition could be formatted like that: https://github.com/ClearcodeHQ/pytest-mysql/blob/master/src/pytest_mysql/executor.py#L118

One for the elasticsearch prior to 2.x the other for later (manual test to confir that 2.x and 5.x are supported and if not a ticket to add such support.